File: backend/app/Models/Reservation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Reservation extends Model
{
    use HasFactory;

    protected $table = 'reservations';

    protected $fillable = [
        'client_id',
        'table_id',
        'time_slot_id',
        'status',
        'code',
        'party_size',
        'date',
        'special_request',
        'phone'
    ];

    public function timeSlot()
    {
        return $this->belongsTo(TimeSlot::class, 'time_slot_id');
    }

    public function table()
    {
        return $this->belongsTo(Table::class, 'table_id');
    }
}

File: mobile/app/src/main/java/com/yehorsk/platea/reservations/domain/models/Reservation.kt
package com.yehorsk.platea.reservations.domain.models

data class Reservation(
    val id: Int,
    val createdAt: String,
    val updatedAt: String,
    val clientId: Int,
    val tableId: Int,
    val timeSlotId: Int,
    val partySize: Int,
    val date: String,
    val status: String,
    val code: String,
    val orderCode: String? = null,
    val specialRequest: String? = null,
    val phone: String,
    val tableNumber: String,
    val startTime: String,
)
